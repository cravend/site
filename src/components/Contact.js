import React from 'react'
import styled from 'styled-components'

import Contact from '../components/contact-card'
import contacts from '../contact.json'

const Flex = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  width: 100%;
  max-width: 100vw;
  margin: 0 auto;
`

export default () => (
  <div>
    <h2>Contact Me</h2>
    <Flex>
      {contacts.map((item, index) => {
        return <Contact key={index} name={item['contact']} url={item['url']} />
      })}
    </Flex>
  </div>
)
