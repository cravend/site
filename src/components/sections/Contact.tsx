import { useTranslations } from "next-intl";

import Link from "../Link";

const Contact = () => {
  const t = useTranslations("sections.Contact");
  return (
    <>
      <h2 id="contact">{t.rich("header")}</h2>
      <ul>
        <li>
          {t.rich("content.phone", {
            Link: (text) => (
              <Link isExternal to="tel:+15138136421">
                {text}
              </Link>
            ),
          })}
        </li>
        <li>
          {t.rich("content.email", {
            Link: (text) => (
              <Link isExternal to="mailto:daltoncraven@proton.me">
                {text}
              </Link>
            ),
          })}
        </li>
        <li>
          {t.rich("content.github", {
            Link: (text) => (
              <Link isExternal to="https://github.com/cravend">
                {text}
              </Link>
            ),
          })}
        </li>
        <li>
          {t.rich("content.gitlab", {
            Link: (text) => (
              <Link isExternal to="https://gitlab.com/cravend">
                {text}
              </Link>
            ),
          })}
        </li>
        <li>
          {t.rich("content.linkedin", {
            Link: (text) => (
              <Link isExternal to="https://www.linkedin.com/in/daltoncraven/">
                {text}
              </Link>
            ),
          })}
        </li>
        <li>
          {t.rich("content.gpg", {
            Link: (text) => (
              <Link isExternal to="https://github.com/cravend.gpg">
                {text}
              </Link>
            ),
          })}
        </li>
      </ul>
    </>
  );
};

export default Contact;
