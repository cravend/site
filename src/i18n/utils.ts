import { locales, defaultLocale, RTL_LOCALES } from "./config";

import type { Locale } from "./config";
import type { NextRouter } from "next/router";

/**
 * @description This function is used to check if a string is a valid locale.
 *
 * @param localeToTest The string to check.
 * @returns True if the string is a valid locale, false otherwise.
 */
export const isLocale = (localeToTest?: string): localeToTest is Locale =>
  locales.some((locale) => locale === localeToTest);

/**
 * @description This function is used to get the current locale.
 *
 * If a custom locale is set by \@next/router, it will be returned.
 * If no custom locale is set, the router's default locale will be returned.
 * If the router's default locale isn't set, the default locale will be returned.
 *
 * @param router The NextRouter instance.
 * @returns The current locale.
 */
export const getLocale = (router: NextRouter | null) => {
  if (!router) return defaultLocale;
  if (isLocale(router.locale)) return router.locale;
  if (isLocale(router.defaultLocale)) return router.defaultLocale;
  return defaultLocale;
};

/**
 * @description This function is used to toggle the current locale.
 *
 * @param locale The current locale to be toggled.
 * @returns The new locale.
 */
export const toggleLocale = (locale: Locale) =>
  locale === locales[0] ? locales[1] : locales[0];

/**
 * @description This function is used to get the text direction of the current locale.
 *
 * @param locale The locale to be checked
 * @returns True if the locale is read right-to-left, false otherwise
 */
export const isRtl = (locale: Locale) => RTL_LOCALES.includes(locale);
