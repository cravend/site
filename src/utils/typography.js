import Typography from 'typography'

const typography = new Typography({
  baseLineHeight: 1.666,
  headerFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  bodyFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  googleFonts: [
    {
      name: 'Poppins',
      styles: ['300', '600', '700']
    }
  ]
})

export default typography
