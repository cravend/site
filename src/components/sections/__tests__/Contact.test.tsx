import { render } from "../../../tests/utils";
import Contact from "../Contact";

describe("<Contact />", () => {
  it("renders", () => {
    expect.assertions(1);
    const { container } = render(<Contact />);
    expect(container).toMatchSnapshot();
  });
});
