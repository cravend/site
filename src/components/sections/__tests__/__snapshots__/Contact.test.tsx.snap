// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> renders 1`] = `
<div>
  <h2
    id="contact"
  >
    Contact
  </h2>
  <ul>
    <li>
      <strong>
        American Phone
      </strong>
      : 
      <a
        class="primary"
        href="tel:+15138136421"
      >
        +1 513 813 6421
      </a>
    </li>
    <li>
      <strong>
        Email
      </strong>
      : 
      <a
        class="primary"
        href="mailto:daltoncraven@proton.me"
      >
        daltoncraven@proton.me
      </a>
    </li>
    <li>
      <strong>
        GitHub
      </strong>
      : 
      <a
        class="primary"
        href="https://github.com/cravend"
      >
        cravend
      </a>
    </li>
    <li>
      <strong>
        GitLab
      </strong>
      : 
      <a
        class="primary"
        href="https://gitlab.com/cravend"
      >
        cravend
      </a>
    </li>
    <li>
      <strong>
        LinkedIn
      </strong>
      : 
      <a
        class="primary"
        href="https://www.linkedin.com/in/daltoncraven/"
      >
        daltoncraven
      </a>
    </li>
    <li>
      <strong>
        GPG
      </strong>
      : 
      <a
        class="primary"
        href="https://github.com/cravend.gpg"
      >
        0xFF3FA75081B3B98C
      </a>
    </li>
  </ul>
</div>
`;
